src/app/ui/comment/comment.component.ts

```diff
@@ -21,11 +23,18 @@ export class CommentComponent implements OnInit {
   allowToMarkAsSolution: boolean;
 
   @Input()
-  topic: TopicId;
+  topic: TopicWithCreatorId;
 
-  constructor(private db: DbService, private snackBar: MatSnackBar) { }
+  isCreator: boolean;
+  isAdmin: Observable<boolean>;
+  canMark: boolean;
+
+  constructor(private db: DbService, private snackBar: MatSnackBar, private auth: AuthService) { }
 
   ngOnInit() {
+    this.isAdmin = this.auth.isAdminForCourse(this.topic.linkedCourseId);
```

Comment:
Your solution is specific to the case the topic is linked to course. What if it is linked to a question?

Reply:
i can see here that even when the topic is linked to a question, it's still linked to the course of the question. so isn't that enough?
![image](https://user-images.githubusercontent.com/19844997/51498262-8d252100-1dce-11e9-965f-335bca99f59a.png)

Reply:
it still works
![image](https://user-images.githubusercontent.com/19844997/51498357-d8d7ca80-1dce-11e9-865e-927fbda447a6.png)

Reply:
If it works then we have another problem and you should find it.. It should not work since topic linked to question has no linkedCourseId. It has only linkedQuestionId.